runtime/src/lib.rs

```diff
@@ -335,47 +335,29 @@ impl_runtime_apis! {
         }
     }
 
+    // Documentation is here: 
+    // https://substrate.dev/rustdocs/master/sp_transaction_pool/runtime_api/trait.TaggedTransactionQueue.html#method.validate_transaction
     impl sp_transaction_pool::runtime_api::TaggedTransactionQueue<Block> for Runtime {
         fn validate_transaction(tx: <Block as BlockT>::Extrinsic) -> TransactionValidity {
             use sp_runtime::traits::Hash;
             
             // Extrinsics representing UTXO transaction need some special handling
-            if let Some(&utxo::Call::execute(ref transaction)) = IsSubType::<utxo::Module<Runtime>, Runtime>::is_sub_type(&tx.function) {
+            if let Some(&utxo::Call::spend(ref transaction)) = IsSubType::<utxo::Module<Runtime>, Runtime>::is_sub_type(&tx.function) {
                 // List of tags to require
                 let requires;
-
-                // Transaction priority to assign
-                let priority;
                 
-                match <utxo::Module<Runtime>>::check_transaction(&transaction) {
-                    // Transaction verification failed
-                    Err(e) => {
-                        sp_runtime::print(e);
-                        return Err(TransactionValidityError::Invalid(InvalidTransaction::Custom(1)));
-                    }
-
-                    // Transaction is valid and verified
-                    Ok(utxo::CheckInfo::Totals {input, output}) => {
-                        // All input UTXOs were found, so we consider input conditions to be met
-                        requires = Vec::new();
+                match <utxo::Module<Runtime>>::has_race_condition(&transaction) {
 
-                        // Priority is based on a transaction fee that is equal to the leftover value
-                        let max_priority = utxo::Value::from(TransactionPriority::max_value());
-                        priority = max_priority.min(input - output) as TransactionPriority;
-                    }
+                    // All input UTXOs were found, so we consider input conditions to be met
+                    None => { requires = Vec::new(); }
```

Comment:
This would also be simplified by removing the `Option<_>` from the return type of `has_race_condition`.

Reply:
oh yeah, will do